### PR TITLE
Order and harmony for properties

### PR DESCRIPTION
**Pull request by enqueue**

### CODE REVIEW

File: java/checkstyle/checkstyle.xml [hunk 1]
```diff
@@ -5,15 +5,26 @@
...
     <module name="RegexpSingleline">
         <property name="format" value="\s+$"/>
         <property name="message" value="Trailing whitespace"/>
     </module>
     <module name="LineLength">
-        <property name="fileExtensions" value="java"/>
         <property name="ignorePattern"
                   value="^ *(((public|private|protected)? (static final|abstract))|@ImportFieldCompletion|@Update|@Field|@AttributeOverride|public (abstract |final )?(class|interface|enum)|assertEquals|[\p{Upper}\p{Digit}_]* *(\(|,))"/>
         <property name="max" value="100"/>
+        <property name="severity" value="warning"/>
```
Comment (wiebeck) [inline java/checkstyle/checkstyle.xml:26]: Why just a warning? I think in java files we can be strict.

File: java/checkstyle/checkstyle.xml [hunk 2]
```diff
@@ -190,4 +207,6 @@
         <property name="lineSeparator" value="lf"/>
         <property name="fileExtensions" value="java, groovy, properties"/>
     </module>
+    <module name="OrderedProperties"/>
```
Comment (wiebeck) [inline java/checkstyle/checkstyle.xml:210]: So this means we need to order properties alphabetically?

File: java/checkstyle/checkstyle.xml [hunk 3]
```diff
@@ -22,6 +33,12 @@
                       value="\A/\*(\r\n|\n) \* Copyright \(c\) \d{4}(-\d{4})? Riege Software( International)?\. All rights reserved\.(\r\n|\n) \* Use is subject to license terms\.(\r\n|\n) \*/"/>
             <property name="message" value="Copyright"/>
         </module>
+        <module name="Regexp">
+            <property name="format" value="\.$"/>
+            <property name="ignoreComments" value="true"/>
+            <property name="illegalPattern" value="true"/>
```
Comment (wiebeck) [inline java/checkstyle/checkstyle.xml:39]: What does this do?

File: java/checkstyle/checkstyle.xml [hunk 4]
```diff
@@ -22,6 +33,12 @@
...
+            <property name="message" value="No trailing dots"/>
```
Comment (wiebeck) [inline java/checkstyle/checkstyle.xml:40]: I don't see any `fileExtensions` property, so does this match _any_ file? What about `.properties` files that contain sentences ending with full stops?

File: java/checkstyle/checkstyle.xml (hunk 3 above)
Comment (enqueue) [reply java/checkstyle/checkstyle.xml:39]: Check for trailing dots, this is copied from the Scope checkstyle configuration.

File: java/checkstyle/checkstyle.xml (hunk 2 above)
Comment (enqueue) [reply java/checkstyle/checkstyle.xml:210]: [yes](https://checkstyle.sourceforge.io/config_misc.html#OrderedProperties)

File: java/checkstyle/checkstyle.xml (hunk 1 above)
Comment (enqueue) [reply java/checkstyle/checkstyle.xml:26]: This is how it is configured for Scope, but I agree.

File: .gitignore [hunk 5]
```diff
@@ -0,0 +1,49 @@
+# Gradle work directory
```
Comment (wiebeck) [inline .gitignore:1]: This `.gitignore` is a bit _too_ generic. ;-)

**Review by wiebeck — APPROVED** [review]

**Pull request merged**